contrib/guix/manifest.scm

```diff
@@ -247,12 +246,13 @@ and abstract ELF, PE and MachO formats.")
     (name "osslsigncode")
     (version "2.5")
     (source (origin
-              (method url-fetch)
```

Comment:
Is there a good reason for using `git-fetch` beyond being more consistent? Seems like this is just less efficient, because we get and hash the entire repository instead of just an archive. Probably all the other cases should be using `url-fetch` instead, or at least with the exception of the sourceware repository?

Reply:
> Probably all the other cases should be using `url-fetch` instead, or at least with the exception of the sourceware repository?

It seems the opposite approach is preferable according to [that](https://lists.gnu.org/archive/html/guix-devel/2020-05/msg00224.html) discussion. Anyway, I'm happy to drop that commit if it is indeed controversial.

Reply:
Can it use a `--depth 1` fetch?

Reply:
No, specifying `depth` is not supported afaict.  I'm ok with moving ahead with this as is. I don't think the reasoning provided in the linked posts really applies here, but would still be good follow best practices.

Reply:
I was reminded of this discussion after reading about the recent XZ backdoor (context [here](https://www.openwall.com/lists/oss-security/2024/03/29/4) and [here](https://boehs.org/node/everything-i-know-about-the-xz-backdoor), I'm sure you've all seen it by now). 

>One portion of the backdoor is *solely in the distributed tarballs*.

Specifically, a small change in the upstream source of `m4/build-to-host.m4`, which:

> injected an obfuscated script from the files committed [in the source tree] to be “executed at the end of configure”.

Many projects that use the autotools build system include autogenerated scripts in their signed release tarballs (so consumers can simply run `./configure`). This makes it hard to compare the contents of these tarballs with their git sources (where I believe changes enjoy more scrutiny). For example, try diffing expat's tarball (which Bitcoin uses in `depends`) with its git repository checked out to the same version. It contains 100k+ lines of unauditable autogenerated code soup.

Critically, in the case of XZ, running `autoreconf -fi` would not have prevented the backdoor from being inserted.

The discussion on the linked guix-devel list did not specifically mention the supply chain attack angle, so I wanted to bring it up here. Going with `git-fetch` for the manifest definitely seems to have been the right choice. It may also be worth exploring whether depends should be able to fetch from git repositories as well or automatically clean up autogenerated files in the preprocess stage to somewhat mitigate this attack vector.